poetapp/explorer-web
# Adds Log in for frost-api accounts

Reverts poetapp/explorer-web#97

Resolves #79 
Resolves #135
Related to #83, #85, #86

### Product Spec
Feature: Users should be able to log in with Poet and authenticate themselves.

As a Poet user, I want to be able to sign in to manage my account.

### Description of Changes

* [x] Installs `frost-client` npm package
* [x] Adds Router, SignIn, LoadingPage, Profile Actions
* [x] Refactor Button/Input/Form slightly to work with Login.
* [x] LoadingPage component
* [x] Login Component (Possibly not necessary)
* [x] SignIn Component
* [x] Adds Helpers from frost-web
* [x] LoginPage Component
* [x] SignIn Reducer
* [x] SignIn Saga
* [x] Add Redux-Persist
* [x] Refactor createPoetStore to work with redux-persist
* [x] Loading Page Reducer
* [x] Added proxy for Frost-client api
* [x] Added profile reducer & saga
* [x] Added small handlerRoutes function to handle auth
* [x] Added a user reducer 
* [x] Unit test for all new reducers/sagas

### To Test:
- Add ?ft=login to url to turn the login page feature on. 
  - Ex. https://deploy-preview-98--poet-explorer.netlify.com/login/?ft=login
- Add ?ft=auth to turn the authorization feature on. 
  - Test by first logging at: https://deploy-preview-98--poet-explorer.netlify.com/login/?ft=login
    - Go back to https://deploy-preview-98--poet-explorer.netlify.com/login/?ft=login
      - This should allow you to see the form(auth turned off)
    - Go to https://deploy-preview-98--poet-explorer.netlify.com/login/?ft=login,auth
      - This should send you to the landing page(will eventually send you to dashboard/profile page)
- Should be able to view a work regardless of feature toggles.


### Important Fix

This PR also includes an unrelated but important fix in the Netlify deploy. See [Fix Netlify](https://github.com/poetapp/explorer-web/pull/131).

### Note

This PR was originally created by @jwicks31 in https://github.com/poetapp/explorer-web/pull/91. That one got accidentally merged before being finished, which triggered an unwanted deploy. It was then reverted.

### PR Review Checklist

* [ ] **Do this changes match the product spec?**
* [ ] **Are there changes in this PR that weren't originally in the product spec?**
* [ ] **Is a feature toggle introduced for this PR? Is it disabled by default?**
* [ ] **Have you tested the changes?**
* [ ] **Is QE review required? If so, has this been reviewed by QE?** (For all new features, review by QE is required. Minor changes which already have passing functional tests may not need a new QE review -- when in doubt, loop in a QE reviewer).
* [ ] **Are there code changes that require new unit tests? If so, are those tests included?** (A separate PR for tests is not appropriate).
* [ ] **Are documentation updates required? If so, does this PR include it?** (Separate documentation PR is generally not appropriate).
* [ ] **Are the added/edited files named appropriately and located correctly in the project directory structure?**

### Warning

Merging to `master` automatically triggers a deploy in production.
